### ISSUE TITLE
[api-extractor] Incorrect indentation after updating to 7.18.0 - indentation lost

### ISSUE THREAD

**Issue by JamesBurnside** [microsoft/rushstack#2799]
## Summary

Some of our API file functions lost their indentation after updating to `7.18.0` (from `7.16.1-pr1916.1`)

![image](https://user-images.githubusercontent.com/2684369/124952217-e302d900-dfc8-11eb-9fef-6cc1b661bce5.png)

## Repro steps

See change here: https://github.com/Azure/communication-ui-library/pull/554/files#diff-ec0dc325bdf92a107d64cbdd0904c5aae39b00aac3236fe2bfe0b88ee332471b

And the definitions of one of the affected functions: https://github.com/Azure/communication-ui-library/blob/542d307497dc3610b66c006cc7419c7abbcb56c8/packages/chat-component-bindings/src/chatThreadSelector.ts#L40

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/api-extractor` version? | 7.18.0 |
| Operating system? | Windows and Linux |
| API Extractor scenario? | .api.md |
| Would you consider contributing a PR? | No |
| TypeScript compiler version? | 4.1.5 |
| Node.js version (`node -v`)? | v14.15.0 |

**Comment by JamesBurnside** [comment]
Might be somewhat related to: #2797

**Comment by octogonz** [comment]
This is a bug with the new support for `import()` type expressions (#1050) not being wired up correctly with the improved indenter (#2796).  Basically an intersection of two recent PRs.

It happens because your compiler is emitting types that were inferred, which come out like this:

**packages/chat-component-bindings/dist/dist-esm/chatThreadSelector.d.ts**
```ts
export declare const chatThreadSelector: import("reselect").OutputParametricSelector<import("@internal/chat-stateful-client").ChatClientState, import("./baseSelectors").ChatBaseSelectorProps, {
    userId: string;
    showMessageStatus: boolean;
    messages: import("@internal/react-components").Message<"chat">[];
}, (res1: string, res2: {
    [key: string]: ChatMessageWithStatus;
}, res3: Date, res4: boolean) => {
    userId: string;
    showMessageStatus: boolean;
    messages: import("@internal/react-components").Message<"chat">[];
}>;
```

The compiler ideally should emit named imports instead of these ugly `import()` expressions. 

However, generally speaking, inferred types will be ugly. They describe the apparent shape of an object, which is often different from how a human would think about it.  Thus if you want your API to be easy to understand, it's not a best practice for public APIs to be defined via type inference (generally speaking -- there are exceptions of course).  

BTW if you are generating API reports, inferred types can also be unstable causing spurious diffs in the API report.  In #1958 we observed that the exact same source code can produce different .d.ts output depending on how the `tsc` command is invoked, and @RyanCavanaugh said this is by design, basically because there does not exist a "normal form" for an inferred TypeScript signature.

To summarize:
- We'll fix this bug.  
- You could work around it by declaring your types explicitly.
- That "workaround" is probably a better practice in most cases.

(I'll add a personal opinion that, if it is difficult to manually author a type expression for an SPI, this suggests that the API isn't expressed clearly and could be improved. 😊)

**Comment by 0x009922** [comment]
Any updates?

**Comment by LinsRock** [comment]
Any updates?
